cmp/form/FormModel.ts

```diff
@@ -274,4 +293,37 @@ export class FormModel extends HoistModel {
             }
         );
     }
+
+    private initPersist({
+        fieldsToPersist = null,
+        path = 'form',
+        ...rootPersistWith
+    }: FormPersistOptions) {
+        const fieldNameMap = fieldsToPersist || Object.keys(this.fields);
+
+        fieldNameMap.forEach(name => {
+            PersistenceProvider.create({
+                persistOptions: {
+                    path: `${path}.fields.${name}.value`,
+                    ...rootPersistWith
+                },
+                target: {
+                    getPersistableState: () => {
+                        return new PersistableState(this.fields[name].value);
+                    },
+                    setPersistableState: ({value}) => {
+                        // There is no metadata on a field to denote it is a date.
+                        // Use a regex matcher to tests for dates and format matches accurately.
```

Comment:
can/should we generalize this?

Reply:
Are you referring to updating the form field to denote its 'type" (e.g. date, number, etc)? to use here, rather than a regex matcher for date-like strings?